### PR TITLE
add an option to verify binlog checksum

### PR DESCRIPTION
**Pull request by holys**
Add an option to verify binlog checksum, and disabled by default, so same behaviour as before.
The user can use `BinlogSyncerConfig.VerifyChecksum` to enable verification.

For MySQL, binlog checksum was introduced since 5.6.2, so before 5.6.2, the value of `FormatDescriptionEvent.ChecksumAlgorithm` is `BINLOG_CHECKSUM_ALG_UNDEF`(undefined).

For MariaDB,
>Checksums are enabled with the binlog_checksum option. Until MariaDB 10.2.1, this was disabled by default. From MariaDB 10.2.1, the option is set to CRC32.

So only when `binlog_checksum = CRC32` and `BinlogSyncerConfig.VerifyChecksum = true` will it works.

@siddontang  @GregoryIan   PTAL

### CODE REVIEW

File: replication/parser.go [hunk 1]
```diff
@@ -23,14 +30,14 @@ type BinlogParser struct {
...
-	useDecimal bool
+	useDecimal     bool
+	verifyChecksum bool
 }
 
 func NewBinlogParser() *BinlogParser {
 	p := new(BinlogParser)
 
 	p.tables = make(map[uint64]*TableMapEvent)
-	// p.stop = make(uint32)
```
Comment (siddontang) [inline replication/parser.go]: why remove it?

Comment (holys) [reply replication/parser.go]: It is not used anywhere. If you want to keep it, I will put it back.

Comment (siddontang) [reply replication/parser.go]: I am very suprised that there is no stop field now but how can it pass compilation before....

Comment (holys) [reply replication/parser.go]: It's commented out before, and now I delete it completely.

File: replication/binlogsyncer.go [hunk 2]
```diff
@@ -71,6 +71,8 @@ type BinlogSyncerConfig struct {
 
 	// maximum number of attempts to re-establish a broken connection
 	MaxReconnectAttempts int
+
+	VerifyChecksum bool
```
Comment (csuzhangxc) [inline replication/binlogsyncer.go]: add some comments (like supporting version) ?

Comment (holys) [reply replication/binlogsyncer.go]: done

**Pull request merged**